src/OfflineWorkflowsSample/OfflineWorkflowsSample/ViewModels/Offline/GenerateMapAreaViewModel.cs

```diff
@@ -0,0 +1,353 @@
+        private bool _includeBasemap = true;
+        private int _maximumLevelOfDetail;
+        private string _selectedMaximumScale;
+        private string _selectedMinimumScale;
+        private int _selectedLevelOfDetail;
+
+        public Map Map
+        {
+            get => _map;
+            set
+            {
+                if (_map != value)
+                {
+                    SetProperty(ref _map, value);
+                    RaisePropertyChanged(nameof(IsMapOnline));
+                    RaiseMapChanged();
+                    RefreshCommands();
+                }
+            }
+        }
+
+        public bool IncludeBasemap
+        {
+            get => _includeBasemap;
+            set
+            {
+                SetProperty(ref _includeBasemap, value);
+                ToggleIncludeBasemap();
+            }
+        }
+
+        public int MaximumLevelOfDetail
+        {
+            get => _maximumLevelOfDetail;
+            set => SetProperty(ref _maximumLevelOfDetail, value);
+        }
+
+        public string SelectedMaximumScale
+        {
+            get => _selectedMaximumScale;
+            set => SetProperty(ref _selectedMaximumScale, value);
+        }
+
+        public string SelectedMinimumScale
+        {
+            get => _selectedMinimumScale;
+            set => SetProperty(ref _selectedMinimumScale, value);
+        }
+
+        public int SelectedLevelOfDetail
+        {
+            get => _selectedLevelOfDetail;
+            set
+            {
+                SetProperty(ref _selectedLevelOfDetail, value);
+                UpdateScales();
+            }
+        }
+
+        public bool IsMapOnline
+        {
+            get => _map.Item is PortalItem;
+        }
+
+        #endregion Properties
+
+        #region Misc. Overhead
+
+        private IWindowService _windowService;
+        private MapViewService _mapViewService;
+
+        public GenerateMapAreaViewModel()
+        {
+            _generateMapAreaCommand = new DelegateCommand(GenerateMapArea, () => IsMapOnline);
+            _navigateToMapAreaCommand = new DelegateCommand(NavigateToMapArea, () => IsMapOnline);
+            _openMapFileCommand = new DelegateCommand(RevealInExplorer, () => !IsMapOnline);
+            _resetMapCommand = new DelegateCommand(() => RaiseMapChanged(true), () => !IsMapOnline);
+        }
+
+        public async Task Initialize(Map map, IWindowService windowService, MapViewService mapViewService)
+        {
+            _windowService = windowService;
+            _mapViewService = mapViewService;
+            Map = map;
+            try
+            {
+                _windowService.SetBusyMessage("Loading map...");
+                _windowService.SetBusy(true);
+
+                await Map.LoadAsync();
+
+                try
+                {
+                    // Set the scale information to the UI
+                    var basemapLayer = Map.AllLayers.OfType<ArcGISTiledLayer>().FirstOrDefault();
```

Comment:
You can look for the base `ImageTiledLayer` class instead, which has TileInfo property and also covers WMTSLayer / WebTiledLayer / OpenStreetLayer / BingMapsLayer.